pkg/client/fake/client.go

```diff
@@ -940,6 +1068,12 @@ func (c *fakeClient) patch(obj client.Object, patch client.Patch, opts ...client
 	if err != nil {
 		return err
 	}
+
+	// otherwise the merge logic in the tracker complains
```

Comment:
So the tracker complains, but the apiserver just ignores them right?

So we need this to end up with the same behavior as the apiserver

And I assume it is also possible with the fake client to patch managedFields with the patch, right? (vs. the obj where we are dropping managedFields here)

Reply:
Updated it to not purge them prior to doing SSA and just return the error + test